src/Orchard.Web/Modules/Orchard.Users/Controllers/AccountController.cs

```diff
@@ -224,27 +236,76 @@ public class AccountController : Controller {
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes",
             Justification = "Exceptions result in password not being changed.")]
         public ActionResult ChangePassword(string currentPassword, string newPassword, string confirmPassword) {
-            ViewData["PasswordLength"] = MinPasswordLength;
+            var membershipSettings = _membershipService.GetSettings();
+            ViewData["PasswordLength"] = membershipSettings.GetMinimumPasswordLength(); ;
 
-            if ( !ValidateChangePassword(currentPassword, newPassword, confirmPassword) ) {
+            if (!ValidateChangePassword(currentPassword, newPassword, confirmPassword)) {
                 return View();
             }
 
+            if (PasswordChangeIsSuccess(currentPassword, newPassword, _orchardServices.WorkContext.CurrentUser.UserName)) {
+                return RedirectToAction("ChangePasswordSuccess");
+            }
+            else {
+                return ChangePassword();
+            }
+        }
+
+        [AlwaysAccessible]
+        public ActionResult ChangeExpiredPassword(string username) {
```

Comment:
Yes, because if you have to change your password because it expired then you'll get to this screen before logging in. I.e. this password change screen is accessible for anonymous users and the POST method needs to be different. Thus the form's target should be different on this screen, while there is also a hint informing you about your password being expired.